### ISSUE TITLE
About Reproduction in Forgetting Differnet Concepts

### ISSUE THREAD

**Issue by TonyZhang-sjtu** [OPTML-Group/Unlearn-Saliency#31]
Hi, when reproducing the results of Salun on forgetting different concepts in UnlearnCanvas, we find some difficulties and I want to ask you about the implementation details and hyper-parameters. From page 20 of UnlearnCanvas(https://arxiv.org/abs/2402.11846), we got the instruction that both mask generation step and train-erase step are trained under 10 epochs.

![Image](https://github.com/user-attachments/assets/0b287871-78e0-4ffc-987a-56a0eab84352)

Therefore, at about Line 43 in generate_mask.py, I changed the range from 1 to 10 and for train-erase.py, we use commane --epoch 10 to set the iterations. The forgetting dataset and remaining dataset are from the data file from generate_dataset.py by default. Our modifications and commands are shown as below:

<img width="957" alt="Image" src="https://github.com/user-attachments/assets/1af86fff-d0be-4686-b2d7-ddc17789fe36" />

<img width="741" alt="Image" src="https://github.com/user-attachments/assets/d77062a0-5b71-4c3d-b0c9-f8ff41d58f71" />

<img width="815" alt="Image" src="https://github.com/user-attachments/assets/5627eb67-89ef-4c93-8b23-84bdffd8c49e" />

We applied Salun method on concept Monet and Jellyfish, but it turns out that for Jellyfish, the results for almost all the concepts are about Architectures, but for Monet, the results are relatively reasonable. So, are there any mistakes on the setting of hyperparameters, training epochs, forgetting dataset ot remaining dataset? Would you please give us some hints on that? Thanks very much!
